Sources/Build/BuildOperation.swift

```diff
@@ -30,6 +30,9 @@ public final class BuildOperation: PackageStructureDelegate, SPMBuildCore.BuildS
 
     /// The closure for loading the package graph.
     let packageGraphLoader: () throws -> PackageGraph
+    
+    /// The closure for evaluating extensions in the package graph.
+    let extensionEvaluator: (PackageGraph) throws -> [ResolvedTarget: [ExtensionEvaluationResult]]
```

Comment:
This isn't an ideal way to call this, but is very much along the lines of the existing `packageGraphLoader`.  The entire BuildOperation layer could use some refactoring.